services/minespace-web/src/components/common/DocumentTable.tsx

```diff
@@ -266,20 +318,47 @@ export const DocumentTable = ({
   const minimalProps = isMinimalView
     ? { size: "small" as SizeType, rowClassName: "ant-table-row-minimal" }
     : null;
-  return showVersionHistory ? (
-    <CoreTable
-      condition={isLoaded}
-      dataSource={documents}
-      columns={columns}
-      expandProps={{
-        childrenColumnName: "versions",
-        matchChildColumnsToParent: true,
-        recordDescription: "version history",
-        rowExpandable: (record) => record.number_prev_versions > 0,
-      }}
-    />
-  ) : (
-    <CoreTable columns={columns} dataSource={documents} {...minimalProps} />
+  const renderCoreTable = () => {
+    let element = (
+      <CoreTable columns={columns}  {...(enableBulkActions
+        ? {
```

Comment:
Can we do all the logic for the table props _before_ trying to render the table? Possibly like how we have minimalProps, have bulkActionsProps, versionProps. Would be nice to reduce the cognitive complexity and repetition here.

Reply:
Excellent call, updated in both core-web and minespace-web as discussed.